### PR TITLE
RFC doc for adding uniqBy to enumerable

### PR DESCRIPTION
**Pull request by lorrocha**

### CODE REVIEW

**Comment by rwjblue** [comment]
Was this proposed to support the efforts in https://github.com/emberjs/ember.js/pull/12875?

**Comment by lorrocha** [comment]
Ah, it is now! I snooped around for an RFC for it and didn't find any either pending or accepted, so I thought I would submit my own. I didn't think to look at the open PRs as I thought, as an api change, it would require an RFC before implementation. Sorry about that!

**Comment by rwjblue** [comment]
No apologies, you are absolutely right! I think this RFC applies to that PR very well.

**Comment by lorrocha** [comment]
I'll update the rfc to remove the specifics to the enumerable mixin!

File: text/0000-implement-uniqby.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  var uniqKeys = arguments;
+  var ret = emberA([]);
+
+  this.forEach((item) => {
+    var options = ret;
+
+    for (var i = 0; i < uniqKeys.length; i++) {
+      var key = uniqKeys[i];
+
+      options = options.filter(iter.apply(options, [key, item[key]]));
```
Comment (stefanpenner) [inline text/0000-implement-uniqby.md]: is this supposed to be `get(item, key)`?

**Comment by stefanpenner** [comment]
This seems reasonable, we should discuss at the next meeting.

File: text/0000-implement-uniqby.md (hunk 1 above)
Comment (lorrocha) [inline text/0000-implement-uniqby.md]: Yes, it is! I'll update it in a bit.

**Comment by stefanpenner** [comment]
This is already in ember, we should be good.

**Pull request closed without merging**